### ISSUE TITLE
Version 6.6 (nightly) Front-page.html template causes unexpected error on Editor main view.

### ISSUE THREAD

**Issue by mjasiukiewicz** [WordPress/gutenberg#62845]
### Description

I'm working on custom block theme and noticed that front-page.html template added to my theme (or any other theme - tested on twentytwentyfour too) causes error on main Editor page.

### Step-by-step reproduction instructions

1. Activate any theme with front-page.html template in `your-theme/templates/` directory.
2. Go to Appearance > Editor.
3. Editor will display error in the preview.

### Screenshots, screen recording, code snippet

![error-editor-frontpage](https://github.com/WordPress/gutenberg/assets/6477903/74a8508f-4e80-428b-b52b-ba7a8a71a85b)

Error:
```
Ku<@wp-includes/js/dist/editor.min.js?ver=0d9718d565440a67e9cd:14:28853
hu@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:60308
xi@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:119597
bs@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:108751
vs@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:108679
gs@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:108540
as@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:105347
ls@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:103902
S@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:130523
T@wp-includes/js/dist/vendor/react-dom.min.js?ver=18.3.1:2:131054
```

### Environment info

- Wordpress 6.6 (latest nightly)
- Any theme with front-page.html (even empty template)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
I tried to test this and wasn't able to replicate using WP Playground with WP Nightly and TT4. To be clear, I created a Front Page template then tried to open the Site Editor without any error appearing. Adding to 6.6 board to be safe though.

**Comment by bgardner** [comment]
@annezazu Try testing with the [Blockify](https://wordpress.org/themes/blockify/) theme, as that appears to be the one in the video above.

**Comment by mjasiukiewicz** [comment]
Seems like something has changed in the recent update and I can't replicate the issue now.

**Comment by t-hamano** [comment]
Is anyone still able to reproduce this issue? I tested it on WP6.6 RC1 and can't reproduce it with either TT4 or Blockify.

**Comment by colorful-tones** [comment]
I'm not able to reproduce this issue. I tested with WordPress 6.6 RC-2 (via the Beta Tester plugin on Local) with the Gutenberg 18.7.1 plugin activated. I tested by duplicating TT4's `templates/page.html` and renaming to `templates/front-page.html`, and I also activated and tested the Blockify theme 1.4.10, which already had an available `front-page.html`. 

I will close this issue out since nobody has been able to reproduce it. However, we can reopen it if the problem resurfaces. Thanks!
